charts/hdfs-k8s/templates/init-fs.yaml

```diff
@@ -0,0 +1,64 @@
+{{- if .Values.init_fs }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{.Release.Name}}-init-fs"
+  labels:
+    app.kubernetes.io/managed-by: {{.Release.Service | quote }}
+    app.kubernetes.io/instance: {{.Release.Name | quote }}
+    helm.sh/chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "{{.Release.Name}}-init-fs"
+      labels:
+        app.kubernetes.io/managed-by: {{.Release.Service | quote }}
+        app.kubernetes.io/instance: {{.Release.Name | quote }}
+        helm.sh/chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    spec:
+      containers:
+        - name: hdfs-init-fs
+          image: crs4/hadoopclient:{{ .Values.global.hadoopVersion }}-ubuntu
+          imagePullPolicy: IfNotPresent
+          env:
+            - name: HADOOP_CONF_DIR
+              value: /etc/hadoop-custom-conf
+          command: ['/post-install/create_tree.sh']
+          args:
+          {{- range .Values.init_fs }}
+            - {{ (printf "%s:%s:%s:%s" .dir .owner .group ( .permissions | default "" )) | quote }}
+          {{- end }}
+          volumeMounts:
+            - name: hdfs-config
+              mountPath: /etc/hadoop-custom-conf
+              readOnly: true
+            - name: hdfs-init-fs
+              mountPath: /post-install
+              readOnly: true
+            {{- if .Values.global.kerberosEnabled }}
```

Comment:
The new hook won't work in the Kerberos scenario, since right after installation the cluster is not yet ready to accept hdfs commands. Currently, the user is expected to run a series of commands to complete the Kerberos configuration after installation (see the relevant test case). Indeed, we could use a post-installation hook to automate that part as well. For now it probably does not hurt to leave this bit here. It might end up working if we manage to automate the Kerberos config part (but this hook would need to be executed after that one).